Automattic/jetpack
# Connection: extract restore connection into the JS package

#### Changes proposed in this Pull Request:
* Extract the "Restore Connection" functionality from Jetpack into the Connection JS package

#### Other information:


#### Jetpack product discussion
1202829334238225-as-1202734545165082

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
1. Connect Jetpack and set invalid blog and user tokens using the "Jetpack Debug -> Broken Token" tool.
2. Go to Jetpack Dashboard, you should see the "Restore Connection" notice, click on it.
3.1 The error notice should disappear
3.2 The "Reconnecting Jetpack" notice on the top right should show up
3.3 After a few seconds, the Dashboard should be replaced with the "Unlock all the amazing features..." screen (user connection screen).
4. Click on "Connect your user account", finish the connection, go back to Jetpack Dashboard and confirm everything is connected correctly.
5. Go to the "Broken Token" and set invalid blog and user tokens.
6. Go to the "Reconnect Modal" page: `/wp-admin/admin.php?page=jetpack#/reconnect`, confirm that the "Reconnect Jetpack" modal shows up.
7. Click "Reconnect Jetpack", the "Reconnecting Jetpack" notice should appear in the top right corner.
8. In a few seconds you should see the user connection screen again. Complete the connection flow and confirm you've been reconnected successfully.
